slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -105,22 +105,32 @@ ::grpc::Status SLAMServiceImpl::GetMap(ServerContext *context,
     return grpc::Status::OK;
 }
 
-SLAMServiceActionMode SLAMServiceImpl::GetActionMode() {
-    // TODO: Add a case for updating. Requires that an apriori
-    // map is detected and loaded. Will be implemented in this ticket:
-    // https://viam.atlassian.net/browse/DATA-114
-
-    // TODO: Change this depending on the outcome of this scope
-    // doc:
-    // https://docs.google.com/document/d/1RsT-c0QOtkMKa-rwUGY0-emUmKIRSaO3mzT1v6MtFjk/edit#heading=h.tcicyojyqi6c
-    if (map_rate_sec.count() == -1) {
-        LOG(INFO) << "Running in localization only mode";
-        return SLAMServiceActionMode::LOCALIZING;
+void SLAMServiceImpl::DetermineActionMode() {
+    // Check if there is an apriori map in the path_to_map directory
+    std::vector<std::string> map_filenames =
+        viam::io::ListSortedFilesInDirectory(path_to_map);
+
+    // Check if there is a *.pbstream map in the path_to_map directory
+    for (auto filename : map_filenames) {
+        if (filename.find(".pbstream") != std::string::npos) {
```

Comment:
[opt] potential split of filtering algorithm to take in a list of strings, a substring and return list of all matching ones. This can then be checked for first, last or any occurrences easily (maybe reused in data filtering as well)